src/plenoptic/simulate/models/portilla_simoncelli.py

```diff
@@ -82,934 +92,849 @@ def __init__(
+                                     for i, r in enumerate(reconstructed_images[:-1])]
+        return reconstructed_images
+
+    def _compute_autocorr(self, coeffs_list: List[Tensor]) -> Tuple[Tensor, Tensor]:
+        """Compute the autocorrelation of some statistics.
 
         Parameters
         ----------
-        ch: torch.Tensor
-        vari: torch.Tensor
-            variance of ch
+        coeffs_list :
+            List (of length s) of tensors of shape (batch, channel, *, height,
+            width), where * is zero or one additional dimensions. Intended use
+            case: magnitude_pyr_coeffs (which is list of length n_scales of 5d
+            tensors, with * containing n_orientations) or reconstructed_images
+            (which is a list of length n_scales+1 of 4d tensors)
 
         Returns
         -------
-        skew: torch.Tensor
-            skew of ch or default value (0)
-        kurtosis: torch.Tensor
-            kurtosis of ch or default value (3)
+        autocorrs :
+            Tensor of shape (batch, channel, spatial_corr_width,
+            spatial_corr_width, s, *) containing the autocorrelation (up to
+            distance ``spatial_corr_width//2``) of each element in
+            ``coeffs_list``, computed independently over all but the final two
+            dimensions.
+        vars :
+            3d Tensor of shape (batch, channel, s, *) containing the variance
+            of each element in ``coeffs_list``, computed independently over all
+            but the final two dimensions.
 
         """
-
-        # Find the skew and the kurtosis of the low-pass residual
-        if vari / self.representation["pixel_statistics"]["var"] > 1e-6:
-            skew = self.__class__.skew(ch, mu=0, var=vari)
-            kurtosis = self.__class__.kurtosis(ch, mu=0, var=vari)
+        if coeffs_list[0].ndim == 5:
+            dims = 's o'
+        elif coeffs_list[0].ndim == 4:
+            dims = 's'
         else:
-            skew = 0
-            kurtosis = 3
+            raise ValueError("coeffs_list must contain tensors of either 4 or 5 dimensions!")
+        acs = [signal.autocorrelation(coeff) for coeff in coeffs_list]
+        var = [signal.center_crop(ac, 1) for ac in acs]
+        acs = [ac/v for ac, v in zip(acs, var)]
+        var = einops.pack(var, 'b c *')[0]
+        acs = [signal.center_crop(ac, self.spatial_corr_width) for ac in acs]
+        acs = torch.stack(acs, 2)
+        return einops.rearrange(acs, f'b c {dims} a1 a2 -> b c a1 a2 {dims}'), var
+
+    def _compute_skew_kurtosis_recon(self, reconstructed_images: List[Tensor], var_recon: Tensor,
+                                     img_var: Tensor) -> Tuple[Tensor, Tensor]:
+        """Computes the skew and kurtosis of each lowpass reconstructed image.
+
+        For each scale, if the ratio of its variance to the original image's
+        pixel variance is below a threshold of 1e-6, skew and kurtosis are
```

Comment:
simple answer is because that hardcoded threshold is from the original matlab code. 1e-6 is the resolution of float32, so I should probably set it to the resolution of whatever the input dtype is